pkg/mic/mic.go

```diff
@@ -730,7 +736,41 @@ func (c *Client) updateUserMSI(newAssignedIDs []aadpodid.AzureAssignedIdentity,
 
 	node, err := c.NodeClient.Get(nodeName)
 	if err != nil {
-		glog.Errorf("Lookup of identities on node %s resulted in error %v", nodeName, err)
+		glog.Warningf("Unable to get node %s while updating user msis. Error %v", nodeName, err)
+
+		if !strings.Contains(err.Error(), "not found") {
+			return
```

Comment:
Looks like in the current code as well as in the new addition we are just swallowing the errors which come from here. Shouldn't we propagate this up ?